test/widgets/action_sheet_test.dart

```diff
@@ -1061,6 +1061,82 @@ void main() {
       });
     });
 
+    group('MarkTopicAsReadButton', () {
+      Future<void> setupToTopicActionSheetWithUnreadMessages(WidgetTester tester, {
+        ZulipStream? channel,
+      }) async {
+        addTearDown(testBinding.reset);
+
+        final effectiveChannel = channel ?? eg.stream();
+        const topicName = TopicName('test topic');
+        final message = eg.streamMessage(stream: effectiveChannel, topic: 'test topic');
+        await testBinding.globalStore.add(eg.selfAccount, eg.initialSnapshot(
+          realmUsers: [eg.selfUser],
+          streams: [effectiveChannel],
+          subscriptions: [eg.subscription(effectiveChannel)]));
+        store = await testBinding.globalStore.perAccount(eg.selfAccount.id);
+        connection = store.connection as FakeApiConnection;
+
+        connection.prepare(json: eg.newestGetMessagesResult(
+          foundOldest: true, messages: [message]).toJson());
+
+        await store.addMessage(message);
+        store.unreads.streams[effectiveChannel.streamId] ??= {};
+        store.unreads.streams[effectiveChannel.streamId]![topicName] ??= QueueList<int>();
+        store.unreads.streams[effectiveChannel.streamId]![topicName]!.add(message.id);
```

Comment:
We shouldn't need to modify `store.unreads.streams` here. A convenient way to add unread messages to the store is through `store.handleEvent(MessageEvent(id: 0, message: message))`. There should be some examples in other tests.